### PR TITLE
Boost layout performance

### PR DESCRIPTION
**Pull request by nstepien**
Since we know all the widths/heights/positions of rows/cells, we can absolutely position the rows, and replace **two** _padding_ divs with very dynamic `height`s, with **one** div with a more static `height`.
This improves layout perf from ~5ms to ~3.1ms on my machine. Might also improve more than layout rendering, not sure.

## Before:
image

## After:

image

### CODE REVIEW

**Comment by nstepien** [comment]
I've also fixed summary rows positioning so they're always at the bottom of the grid, even if we don't have enough rows to fill the available space.

## Before:
image
## After:
image

File: src/Row.tsx [hunk 1]
```diff
@@ -47,6 +48,7 @@ function Row<R, SR = unknown>({
       onDragEnter={wrapEvent(handleDragEnter, onDragEnter)}
       onDragOver={wrapEvent(handleDragOver, onDragOver)}
       onDrop={wrapEvent(preventDefault, onDrop)}
+      style={{ top }}
```
Comment (qili26) [inline src/Row.tsx:51]: Just 1 concern here. Let's say if the user has their own `RowRenderer` and they forgot to add this `top` to their wrapper. The whole layout would be messed up, right?

Comment (nstepien) [reply src/Row.tsx:51]: Yuyp. If you pass `top` back to `Row`, or to your wrapper, you'll be fine though.

Comment (qili26) [reply src/Row.tsx:51]: To be honest, I don't know if this worth it given by 5ms to 3ms improvement. As it makes the `RowRenderer` error prone. And I bet people will create issues for this case when they use `CustomRowRenderer`.

Comment (nstepien) [reply src/Row.tsx:51]: Aren't custom row renderers already error-prone?

**Comment by amanmahajan7** [comment]
@nstepien can you catchup?

**Comment by nstepien** [comment]
@amanmahajan7 Done.

**Pull request merged**